include/nbl/video/IDescriptorPool.h

```diff
@@ -29,12 +33,156 @@ class NBL_API IDescriptorPool : public core::IReferenceCounted, public IBackendO
             uint32_t count;
         };
 
-        explicit IDescriptorPool(core::smart_refctd_ptr<const ILogicalDevice>&& dev, uint32_t _maxSets) : IBackendObject(std::move(dev)), m_maxSets(_maxSets) {}
+    private:
+        struct SCombinedImageSampler
+        {
+            core::smart_refctd_ptr<video::IGPUImageView> view;
+            core::smart_refctd_ptr<video::IGPUSampler> sampler;
+        };
+
+        // This construct makes it so that we don't trigger T's constructors and destructors.
+        // TODO(achal): Should we move it to the core namespace?
```

Comment:
why not, make it a separate header just for this then

Reply:
Resolved.